interfaces/builtin/pipewire.go

```diff
@@ -0,0 +1,138 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	"github.com/snapcore/snapd/snap"
+)
+
+const pipewireSummary = `allows access to the pipewire sockets, and offer them`
+
+const pipewireBaseDeclarationSlots = `
+  pipewire:
+    allow-installation:
+      slot-snap-type:
+        - app
+        - core
+    deny-auto-connection: true
```

Comment:
Maybe consider adding:

```
    deny-connection:
      on-classic: false
```

... to match the policy of audio-record/pulseaudio.

Reply:
I don't understand what does that mean... allow to connect a plug to that slot for classic snaps?

BTW: I realized that this is the patch for upstream snapd... but it should be still a draft, because the first one to be merged should be the one in our core-desktop snapd... https://github.com/canonical/ubuntu-core-desktop-snapd/pull/34

I don't know why I changed it from "draft" to "ready"... maybe I mistook both repos...

Reply:
The intent is to disallow connection if the slot is provided by an application snap as opposed to an implicit slot. So a snap wanting to provide the slot on an Ubuntu Core system would need a snap-declaration from the store allowing connection.

It is a bit of a round-about way of saying so, since it's possible to install the same application snap on a classic system.

Reply:
Oh, ok. Thanks!